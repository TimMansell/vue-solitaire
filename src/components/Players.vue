<template>
  <div>
    Players:
    <Counter data-test="player-count" :number="playerCount" />
    (<span :title="playersOnline" data-test="online-title">
      <Counter data-test="online-count" :number="onlinePlayerCount" /> </span
    >)
  </div>
</template>

<script>
import { mapGetters } from 'vuex';
import Counter from '@/components/Counter.vue';

export default {
  name: 'Players',
  components: {
    Counter,
  },
  computed: {
    ...mapGetters(['playerCount', 'onlinePlayerCount']),
    playersOnline() {
      const { onlinePlayerCount } = this;

      const player = onlinePlayerCount > 1 ? 'players' : 'player';

      return `${onlinePlayerCount} ${player} online`;
    },
  },
};
</script>
