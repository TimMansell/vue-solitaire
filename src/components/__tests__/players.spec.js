import { shallowMount } from '@vue/test-utils';
import Players from '@/components/Players.vue';

describe('Players.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(Players, {
      computed: {
        playerCount: () => 1,
        onlinePlayerCount: () => 1,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should show 1 player online in title', () => {
    const wrapper = shallowMount(Players, {
      computed: {
        playerCount: () => 1,
        onlinePlayerCount: () => 1,
      },
    });

    expect(
      wrapper.find('[data-test="online-title"]').attributes().title
    ).toContain('1 player online');
  });

  it('should show 2 players online in title', () => {
    const wrapper = shallowMount(Players, {
      computed: {
        playerCount: () => 1,
        onlinePlayerCount: () => 2,
      },
    });

    expect(
      wrapper.find('[data-test="online-title"]').attributes().title
    ).toContain('2 players online');
  });
});
