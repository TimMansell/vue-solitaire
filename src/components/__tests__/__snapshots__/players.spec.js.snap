// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Players.vue matches snapshot 1`] = `
<div>
  Players:
  <counter-stub number="1" data-test="player-count"></counter-stub>
  (<span title="1 player online" data-test="online-title"><counter-stub number="1" data-test="online-count"></counter-stub></span>)
</div>
`;
